consensus/istanbul/backend/announce.go

```diff
@@ -419,3 +516,125 @@ func (sb *Backend) regossipIstAnnounce(msg *istanbul.Message, payload []byte, an
 
 	return nil
 }
+
+// ===============================================================
+//
+// define the AnnounceVersion messge format, the getAnnounceVersions send and handle function, and the announceVersions handle function
+
+type announceVersion struct {
+	ValAddress           common.Address
+	AnnounceMsgTimestamp uint
+}
+
+// EncodeRLP serializes av into the Ethereum RLP format.
+func (av *announceVersion) EncodeRLP(w io.Writer) error {
+	return rlp.Encode(w, []interface{}{av.ValAddress, av.AnnounceMsgTimestamp})
+}
+
+// DecodeRLP implements rlp.Decoder, and load the ee fields from a RLP stream.
+func (av *announceVersion) DecodeRLP(s *rlp.Stream) error {
+	var msg struct {
+		ValAddress           common.Address
+		AnnounceMsgTimestamp uint
+	}
+
+	if err := s.Decode(&msg); err != nil {
+		return err
+	}
+	av.ValAddress, av.AnnounceMsgTimestamp = msg.ValAddress, msg.AnnounceMsgTimestamp
+	return nil
+}
+
+// The stringer function for the announceVersion type
+func (av *announceVersion) String() string {
+	return fmt.Sprintf("{ValAddress: %s, AnnounceMsgTimstamp: %d}", av.ValAddress.String(), av.AnnounceMsgTimestamp)
+}
+
+// This function will send a GetAnnounceVersions message to a specific peer to request it's announceVersion set
```

Comment:
Nit: The following functions should replace "this function" with the function name according to Effective Go.